spec/formatting.md

```diff
@@ -168,28 +204,47 @@ Its resolved value is defined by _literal resolution_.
 
 ### Literal Resolution
 
-The resolved value of a _text_ or a _literal_ is
+The _resolved value_ of a _text_ or a _literal_ contains
 the character sequence of the _text_ or _literal_
 after any character escape has been converted to the escaped character.
 
 When a _literal_ is used as an _operand_
 or on the right-hand side of an _option_,
-the formatting function MUST treat its resolved value the same
+the formatting function MUST treat its _resolved value_ the same
 whether its value was originally _quoted_ or _unquoted_.
 
 > For example,
 > the _option_ `foo=42` and the _option_ `foo=|42|` are treated as identical.
 
-The resolution of a _text_ or _literal_ MUST resolve to a string.
+The resolution of a _text_ or _literal_ MUST resolve with a string value.
+
+> For example, in a JavaScript formatter
+> the _resolved value_ of a _text_ or a _literal_ could have the following implementation:
+>
+> ```ts
+> class MessageLiteral implements MessageValue {
+>   constructor(value: string) {
+>     this.formatToString = () => value;
+>     this.getValue = () => value;
+>   }
+>   resolvedOptions: () => ({});
+>   selectKeys(_keys: string[]) {
+>     throw Error("Selection on unannotated literals is not supported");
+>   }
+> }
+> ```
 
 ### Variable Resolution
 
 To resolve the value of a _variable_,
 its _name_ is used to identify either a local variable or an input variable.
-If a _declaration_ exists for the _variable_, its resolved value is used.
+If a _declaration_ exists for the _variable_, its _resolved value_ is used.
 Otherwise, the _variable_ is an implicit reference to an input value,
 and its value is looked up from the _formatting context_ _input mapping_.
 
+An implemementation MAY choose to wrap or otherwise represent all input values
+using the same representation it uses for all _resolved values_.
```

Comment:
Highlighting this addition, which might otherwise be lost among the _resolved value_ link additions.